### PR TITLE
Centralize surface route normalization

### PR DESCRIPTION
**Pull request by pbakaus**
## Summary
- centralize route validation and canonicalization for explicit route identifiers and inferred absolute-style routes
- preserve existing surface-brief paths, frontmatter, resolution reasons, and public exports
- characterize equivalent query-bearing, repeated-slash, traversal, and root-route inputs

## Hindsight judgment
Hindsight is 20/20: if these requirements were implemented today, the two accepted route input forms should delegate to one durable normalization rule. The prior implementation duplicated query stripping, slash collapsing, trailing-slash handling, and traversal validation across branches. This change introduces one private helper and keeps file and URL handling separate.

## Before / after
- Primary file: 151 lines to 149 lines
- Route canonicalization implementations: 2 to 1
- Public exports and entry points: unchanged
- Responsibilities: unchanged; normalizeSurfaceTarget still classifies URL, route, and project-file targets, while the private helper now owns route syntax
- Total production source touched: one file, net minus 2 lines

## Validation
- node --test tests/surface-brief.test.mjs: 6 passed
- node --test tests/context.test.mjs tests/staleness.test.mjs: 164 passed
- bun run build: passed
- bun run test: passed, including core, detector/browser, Live, framework, and plugin E2E suites

## Delivery notes
AI assistance: OpenAI Codex prepared this PR under pbakaus scheduled architecture-simplification authorization.

Generated provider output was intentionally omitted; the source-first build completed without tracked generated churn.

Remaining risk is low and limited to route classification edge cases. Characterization covers both accepted forms and traversal rejection, and the full suite is green.

---

> [!NOTE]
> **Low Risk**
> Internal refactor of route classification with characterization tests; public exports and brief paths are unchanged.
> 
> **Overview**
> **Route targets** (`route:/…` and absolute paths that are not project files) now share one private `normalizeRouteTarget` helper instead of duplicating query/hash stripping, slash collapsing, trailing-slash trimming, and `..` rejection in two branches of `normalizeSurfaceTarget`.
> 
> Explicit `route:` inputs, bare `/`, and inferred routes when the path is not a resolvable project file all delegate to that helper. The extra `!trimmed.includes('..')` guard on the inferred-route branch is dropped because traversal is rejected inside the helper.
> 
> A new test asserts that equivalent messy inputs (`route:/docs//intro/?from=nav#top` vs `/docs//intro/…`) canonicalize to `route:/docs/intro`, and that `..` segments return `null` for both forms.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 4f793678f2e3259ec4402f846cf8b6cfe486131c. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: skill/scripts/lib/surface-briefs.mjs [hunk 1]
```diff
@@ -21,21 +27,13 @@ export function normalizeSurfaceTarget(target, { projectRoot = process.cwd() } =
...
+  if (trimmed === '/') return normalizeRouteTarget(trimmed);
   if (trimmed.startsWith('/')) {
     const absolute = path.resolve(trimmed);
     const relativeToProject = path.relative(projectRoot, absolute);
     const isProjectFile = relativeToProject && !relativeToProject.startsWith('..') && !path.isAbsolute(relativeToProject);
-    if (!isProjectFile && !fs.existsSync(absolute) && !trimmed.includes('..')) {
-      const normalizedRoute = trimmed.split(/[?#]/, 1)[0].replace(/\/{2,}/g, '/').replace(/\/$/, '') || '/';
-      return `route:${normalizedRoute}`;
-    }
+    if (!isProjectFile && !fs.existsSync(absolute)) return normalizeRouteTarget(trimmed);
```
[the remark this answers is not in the document]
Comment (pbakaus) [reply skill/scripts/lib/surface-briefs.mjs:36]: Won't fix: this behavior predates the PR and is intentionally preserved. Bare // and /// remain filesystem-root syntax; only literal / is the characterized inferred root route, while route:// explicitly opts into route syntax. Accepting new bare forms would change the public input contract and violate this refactor's behavior-preserving scope.

**Comment by pbakaus** [comment]
> Re: comment 5308167994

Won't fix: the summary describes a pre-existing slash-only input distinction. Expanding bare // and /// into accepted route forms would be behavior change outside this refactor; the actionable inline thread records the full rationale and is resolved.
